libraries/AP_Common/AP_Common.cpp

```diff
@@ -36,3 +36,19 @@ bool is_bounded_int32(int32_t value, int32_t lower_bound, int32_t upper_bound)
     return false;
 
 }
+
+/**
+ * return the numeric value of an ascii hex character
+ *
+ * @param [in] a hexa character
+ * @return numeric value
+ */
+ int16_t char_to_hex(int8_t a)
```

Comment:
This is a very strange return type.  I wonder if this was originally intended to return `-1` on error?

Regardless, this should be `uint8_t`, right?

Reply:
I know this implementation isn't yours, but it could be made more efficient.  10 times in 16 the LAST clause here will be taken - meaning we've done more comparisons than needed.  Something to fix later, perhaps.

Reply:
... also, not a particularly apt name.

How about `hexchar_to_int`, or possibly just `unhex` (`hex` being what you use to go the other way in a lot of languages)

Reply:
@peterbarker san.... thanks Comments.
1. Change the name to unhex.
2. Change the method type to uint8_t.
3. Let's make the process easy. You can reduce the same process if you use lower case letters and upper case letters. Such. .